core-bundle/src/Image/Studio/Figure.php

```diff
@@ -24,8 +24,10 @@
  * use the provided legacy helper methods to manually apply the data to them.
  *
  * Wherever possible, the actual data is only requested/built on demand.
+ *
+ * @final
  */
-final class Figure
+class Figure
```

Comment:
Perfect example why classes should only be `final` if they implement an interface. 🤓 See https://tomasvotruba.com/blog/2019/03/28/how-to-mock-final-classes-in-phpunit/

Reply:
I personally go for the strategy: Use `final` everywhere until you (or your users) have got a use case for it to make the API surface as small as possible. This is where I disagree with the article: I do not need an interface until then and the need for mocking does not justify adding one.

Using `class final` should IMHO be preferred to `/** @final  */` because it's more strict, but it basically says the same: Do not inherit from this class and expect things to work. If mocking is helpful for a test, I do not feel bad to switch to the annotation. :smile: There are concepts in other programming languages ("friends") that would allow to be more specific while still being strict about the API but PHP doesn't have these (yet?).